### ISSUE TITLE
vcd_org_vdc - unable to update storage profile limit in-place

### ISSUE THREAD

**Issue by nilic** [vmware/terraform-provider-vcd#461]
### Terraform Version
Terraform v0.12.21
+ provider.vcd v2.6.0
vCD 9.7.0.15498291

### Affected Resource(s)
- vcd_org_vdc

### Terraform Configuration Files
```hcl
resource "vcd_org_vdc" "my_vdc" {
  name = "bugtest_vdc"
  org  = "MyOrg"

  allocation_model  = "ReservationPool"
  network_pool_name = "MyNPNP"
  provider_vdc_name = "MyPVDC"

  compute_capacity {
    cpu {
      allocated = 2048
    }

    memory {
      allocated = 2048
    }
  }

  storage_profile {
    name    = "Gold"
    limit   = 200
    default = true
  }

  enabled                  = true
  enable_thin_provisioning = true
  enable_fast_provisioning = false
  delete_force             = true
  delete_recursive         = true
}
```

After applying the above config file, I have changed the storage profile limit from 200 to 300 and tried to apply the change.

### Debug Output
Decided to paste the relevant part of debug output here since it's not too long.

```
vc2020/03/01 22:55:38 [DEBUG] vcd_org_vdc.my_vdc: applying the planned Update change
d2020-03-01T22:55:38.392+0100 [DEBUG] plugin.terraform-provider-vcd_v2.6.0_x4.exe: 2020/03/01 22:55:38 [TRACE] vdc update initiated
_org_vdc.my_vdc: Modifying... [id=urn:vcloud:vdc:3b156b88-6bcb-485a-8c7e-296e5dc97784]
2020-03-01T22:55:45.091+0100 [DEBUG] plugin.terraform-provider-vcd_v2.6.0_x4.exe: 2020/03/01 22:55:45 [TRACE] adding/updating metadata to VDC
2020-03-01T22:55:45.585+0100 [DEBUG] plugin.terraform-provider-vcd_v2.6.0_x4.exe: 2020/03/01 22:55:45 [TRACE] vdc update completed: bugtest_vdc
2020-03-01T22:55:45.585+0100 [DEBUG] plugin.terraform-provider-vcd_v2.6.0_x4.exe: 2020/03/01 22:55:45 [TRACE] vdc read initiated
2020/03/01 22:55:46 [WARN] Provider "registry.terraform.io/-/vcd" produced an unexpected new value for vcd_org_vdc.my_vdc, but we are tolerating it because it
 is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
vcd_org_vd      - .storage_profile[0].limit: was cty.NumberIntVal(300), but now cty.NumberIntVal(200)
c.my_vdc: Modifications complete after 8s [id=urn:vcloud:vdc:3b156b88-6bcb-485a-2020-03-01T22:55:46.454+0100 [DEBUG] plugin.terraform-provider-vcd_v2.6.0_x4.e
xe: 2020/03/01 22:55:46 [TRACE] vdc read completed: &types.AdminVdc{Xmlns:"http://www.vmware.com/vcloud/v1.5", Vdc:types.Vdc{HREF:"https://myvcd/a
pi/admin/vdc/3b156b88-6bcb-485a-8c7e-296e5dc97784", Type:"application/vnd.vmware.admin.vdc+xml", ID:"urn:vcloud:vdc:3b156b88-6bcb-485a-8c7e-296e5dc97784", Ope
rationKey:"", Name:"bugtest_vdc", Status:1, Link:types.LinkList{(*types.Link)(0xc00085b810), (*types.Link)(0xc00085b860), (*types.Link)(0xc00085b8b0), (*types
.Link)(0xc00085b900), (*types.Link)(0xc00085b950), (*types.Link)(0xc00085b9a0), (*types.Link)(0xc00085b9f0), (*types.Link)(0xc00085ba40), (*types.Link)(0xc000
85ba90), (*types.Link)(0xc00085bae0), (*types.Link)(0xc00085bb30), (*types.Link)(0xc00085bb80), (*types.Link)(0xc00085bbd0), (*types.Link)(0xc00085bc20), (*ty
pes.Link)(0xc00085bc70), (*types.Link)(0xc00085bcc0), (*types.Link)(0xc00085bd10), (*types.Link)(0xc00085bd60), (*types.Link)(0xc00085bdb0)}, Description:"",
AllocationModel:"ReservationPool", ComputeCapacity:[]*types.ComputeCapacity{(*types.ComputeCapacity)(0xc000749fe0)}, ResourceEntities:[]*types.ResourceEntitie
s{(*types.ResourceEntities)(0xc0001b9160)}, AvailableNetworks:[]*types.AvailableNetworks{(*types.AvailableNetworks)(0xc0001b9200)}, Capabilities:[]*types.Capa
bilities{(*types.Capabilities)(0xc0000066b8)}, NicQuota:0, NetworkQuota:0, VMQuota:0, IsEnabled:true, Tasks:(*types.TasksInProgress)(nil), UsedNetworkCount:0,
 VdcStorageProfiles:(*types.VdcStorageProfiles)(0xc0001b9a20)}, ResourceGuaranteedMemory:(*float64)(0xc000858de0), ResourceGuaranteedCpu:(*float64)(0xc000858d
f8), VCpuInMhz:(*int64)(0xc000858e20), IsThinProvision:(*bool)(0xc000858e39), NetworkPoolReference:(*types.Reference)(0xc000826100), ProviderVdcReference:(*ty
pes.Reference)(0xc0008262c0), UsesFastProvisioning:(*bool)(0xc000858eeb), OverCommitAllowed:false, VmDiscoveryEnabled:false}
8c7e-296e5dc97784]
2020-03-01T22:55:46.603+0100 [DEBUG] plugin: plugin process exited: path=E:\somepath\terraform-provide
r-vcd_v2.6.0_x4.exe pid=12460
2020-03-01T22:55:46.603+0100 [DEBUG] plugin: plugin exited

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

This is the debug output after changing the limit from 200 to 300 and performing `terraform apply`

### Expected Behavior
Storage limit for the "Gold" storage profile should be changed from 200 to 300.

### Actual Behavior
Storage limit for the "Gold" storage profile stays the same. `terraform apply` exited with success but upon reviewing the limit from the UI it is visible that the limit hasn't changed. Also, another run of `terraform plan` offers to perform the same change again.

```
...
      ~ storage_profile {
            default = true
            enabled = true
          ~ limit   = 200 -> 300
            name    = "Gold"
        }
...
Plan: 0 to add, 1 to change, 0 to destroy.
```

### Steps to Reproduce
1. create an Org vDC
2. change the value for `limit` inside the `storage_profile` block
3. apply changes

**Comment by vbauzysvmware** [comment]
Hi @nilic Thank you for noticing that. After evaluation I concluded that isn't possible to update storage profile/details with same API. To manage storage profiles of VDC it is needed to make additional call to `api/admin/vdcStorageProfile/{id}`. There isn't easy/fast fix for now as we need to add support of that in govcd.

**Comment by vbauzysvmware** [comment]
fixed and will be released in 3.1 version
